### PR TITLE
feat: add bathroom filter

### PR DESCRIPTION
**Pull request by chriscasto**
# Pull Request Template

## Issue Overview

- [X] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR adds a filter on the number of bathrooms in any of the units on a listing.  It supports numeric comparison operations, so it can be used to respond to a request for "listings with units that have greater than or equal to 2 bathrooms".

## How Can This Be Tested/Reviewed?

There is an e2e test under `listings-combined.e2e-spec.ts`, same as with the other filters.  Just run `yarn test:e2e:local` or `yarn test:e2e:local -t combined` (faster) to verify test results.  You can also run `yarn test` to run unit tests if you want to be thorough.

## Checklist:

- [X] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [X] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have assigned reviewers
- [X] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [X] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend/core/types/src/backend-swagger.ts [hunk 1]
```diff
@@ -700,6 +700,21 @@ export class AssetsService {
       axios(configs, resolve, reject)
     })
   }
+  /**
+   * Upload asset
+   */
+  upload(options: IRequestOptions = {}): Promise<Asset> {
```
Comment (isaac-looker) [inline backend/core/types/src/backend-swagger.ts:706]: is this related to the bathrooms filtering?

**Review by phattya — APPROVED** [review]
Lgtm! thanks!

File: backend/core/types/src/backend-swagger.ts (hunk 1 above)
Comment (chriscasto) [reply backend/core/types/src/backend-swagger.ts:706]: It came up when I regenerated the client, which I guess was using a different version of the backend.  I suspect it was for PR #96 which just got merged, so it should accurately reflect the backend state.
